app/src/main/java/com/example/harvestfresh/fragments/CartFragment.java

```diff
@@ -108,9 +109,10 @@ private void cartCheckout(){
         btnConfirm.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Log.d(TAG, "Buy");
+                for (int i = 0; i < allCarts.size(); i++) {
```

Comment:
Try using a foreach loop here
for (Cart cart : allCarts) {
cart.deleteInBackground();
}

Its best to avoid an indexed for loop unless youre specifically using the index, because it risks introducing off-by-one bugs and your runtime can optimize more for foreach loops. Its also more legible.